main/src/ca/uwaterloo/flix/tools/pkg/MavenPackageManager.scala

```diff
@@ -15,25 +15,67 @@
  */
 package ca.uwaterloo.flix.tools.pkg
 
+import ca.uwaterloo.flix.tools.pkg.Dependency.MavenDependency
 import ca.uwaterloo.flix.util.Result
+import ca.uwaterloo.flix.util.Result.{Err, Ok, ToOk}
 
 import java.io.PrintStream
 import coursier._
 
 object MavenPackageManager {
 
-  def installDeps(manifest: Manifest)(implicit out: PrintStream): Result[Unit, ManifestError] = {
+  val scalaVersion = "2.13"
 
+  /**
+    * Installs all MavenDependencies for a Manifest,
+    * including transitive dependencies using coursier.
+    */
+  def installAll(manifest: Manifest)(implicit out: PrintStream): Result[List[String], PackageError] = {
+    val depStrings = getMavenDependencyStrings(manifest)
 
-    val resolution = Resolve()
-      .addDependencies(dep"org.tpolecat:doobie-core_2.12:0.6.0")
-      .run()
-    println(resolution)
+    createCoursierDependencies(depStrings).flatMap { deps =>
+      val installed = try {
+        val res = deps.foldLeft(Resolve())((res, dep) => res.addDependencies(dep))
+        val resolution = res.run()
 
-    ???
+        val fetch = resolution.dependencies.foldLeft(Fetch())((f, dep) => {
+          out.println(s"Installing ${dep.module.toString()}");
+          f.addDependencies(dep)
+        })
+        fetch.run()
 
+        resolution.dependencies.map(dep => dep.module.toString()).toList
+      } catch {
+        case e: Exception =>
+          val message = e.getMessage.replaceAll("[^a-zA-Z0-9:. ]", "/").split('/').apply(0)
```

Comment:
What's this? Maybe add a comment?

Reply:
That is because the error message  spans multiple lines and I just want the name. For some reason splitting on '\n' made the test I wrote for this behave very oddly, so I decided to only allow certain characters in the string to make it easy to manipulate